### PR TITLE
Add blocks sounds

### PR DESCRIPTION
**Pull request by BeYkeRYkt**
https://github.com/SpaceManiac/Glowstone/issues/99

### CODE REVIEW

**Comment by kitskub** [comment]
@BeYkeRYkt 

You don't need to tag your changes with a comment with your name. Just make the necessary changes. And if you remove old code, just delete it. Don't just comment  it out.

**Comment by BeYkeRYkt** [comment]
@kitskub 
Okay, next time I will consider, thank.

**Comment by kitskub** [comment]
@BeYkeRYkt 

Looks like you changed from spaces to tabs. (Maybe the other way around) Make sure you follow what the rest of the file is doing.

**Comment by SpaceManiac** [comment]
First off, thanks for the contribution! I appreciate the effort you went to to determine all the different blocks' place sounds. A few concerns, I hope they don't seem overwhelming:
- Keep an eye on spaces/tabs and unneeded whitespace or imports changes. I can fix them on my end but it's always nice not to have to (I'm aware an actual style guide is currently lacking).
- What do the changes to ItemSign accomplish? The playSound is called by the appropriate BlockType's rightClickBlock method so the state updates seem odd.
- I think I would prefer setPlaceSound() returning BlockType for chaining rather than having reg() change to return a BlockType.
- `place_sound` should be `placeSound`.
- `SAND` does not need to be `BlockDirectDrops`. The default `BlockType` handles its drops sufficiently.

File: src/main/java/net/glowstone/block/blocktype/BlockType.java [hunk 1]
```diff
@@ -146,8 +147,8 @@ public final void rightClickBlock(GlowPlayer player, GlowBlock against, BlockFac
 
         // play a sound effect
         // todo: vary sound effect based on block type
-        target.getWorld().playSound(target.getLocation(), Sound.DIG_WOOD, 1, 1);
-
+        target.getWorld().playSound(target.getLocation(), place_sound, 1, 0.8F);
+
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockType.java:151]: Extra whitespace

File: src/main/java/net/glowstone/block/itemtype/ItemSign.java [hunk 2]
```diff
@@ -13,15 +12,14 @@
...
     public void rightClickBlock(GlowPlayer player, GlowBlock target, BlockFace face, ItemStack holding, Vector clickedLoc) {
-        BlockType placeAs;
         if (face == BlockFace.UP) {
-            placeAs = ItemTable.instance().getBlock(Material.SIGN_POST);
+            setPlaceAs(ItemTable.instance().getBlock(Material.SIGN_POST));
         } else if (face == BlockFace.DOWN) {
             return;
         } else {
-            placeAs = ItemTable.instance().getBlock(Material.WALL_SIGN);
+        	setPlaceAs(ItemTable.instance().getBlock(Material.WALL_SIGN));
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/itemtype/ItemSign.java:20]: I spy a tab

File: src/main/java/net/glowstone/block/itemtype/ItemType.java [hunk 3]
```diff
@@ -136,4 +136,9 @@ public void rightClickBlock(GlowPlayer player, GlowBlock target, BlockFace face,
     public final String toString() {
         return getClass().getSimpleName() + "{" + getId() + " -> " + getMaterial() + "}";
     }
+    
+
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/itemtype/ItemType.java:140]: Extra white space

Edit: Note: 2 lines and extra spaces

File: src/main/java/net/glowstone/entity/GlowHumanEntity.java [hunk 4]
```diff
@@ -302,7 +304,7 @@ public InventoryView openEnchanting(Location location, boolean force) {
         /*InventoryView view = new GlowInventoryView(this, new GlowEnchantInventory() ...);*/
         return null;
     }
-
+
```
Comment (turt2live) [inline src/main/java/net/glowstone/entity/GlowHumanEntity.java:307]: Extra white space

File: src/main/java/net/glowstone/block/ItemTable.java [hunk 5]
```diff
@@ -37,77 +39,82 @@ private ItemTable() {}
...
+        reg(Material.REDSTONE, new ItemPlaceAs(Material.REDSTONE_WIRE)).setPlaceSound(Sound.DIG_STONE);
+        reg(Material.SUGAR_CANE, new ItemPlaceAs(Material.SUGAR_CANE_BLOCK)).setPlaceSound(Sound.DIG_GRASS);
+        reg(Material.DIODE, new ItemPlaceAs(Material.DIODE_BLOCK_OFF)).setPlaceSound(Sound.DIG_STONE);
+        reg(Material.BREWING_STAND_ITEM, new ItemPlaceAs(Material.BREWING_STAND)).setPlaceSound(Sound.DIG_STONE);
+        reg(Material.CAULDRON_ITEM, new ItemPlaceAs(Material.CAULDRON)).setPlaceSound(Sound.DIG_STONE);
+        reg(Material.FLOWER_POT_ITEM, new ItemPlaceAs(Material.FLOWER_POT)).setPlaceSound(Sound.DIG_STONE);
+        reg(Material.SKULL_ITEM, new ItemPlaceAs(Material.SKULL)).setPlaceSound(Sound.DIG_STONE);
+        reg(Material.REDSTONE_COMPARATOR, new ItemPlaceAs(Material.REDSTONE_COMPARATOR_OFF)).setPlaceSound(Sound.DIG_STONE);
+        
+        reg(Material.SAND, new BlockDirectDrops(Material.SAND)).setPlaceSound(Sound.DIG_SAND);
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/ItemTable.java:100]: Why isn't sand with it's buddies?

**Comment by turt2live** [comment]
@BeYkeRYkt Your PR is currently not mergable. Please update it as required.

**Comment by turt2live** [comment]
@BeYkeRYkt Once this compiles it can be further reviewed. It currently has competition with #242 which is more active and does a slightly cleaner implementation. Are you still working on this?

File: src/main/java/net/glowstone/block/ItemTable.java [hunk 6]
```diff
@@ -37,77 +39,82 @@ private ItemTable() {}
...
-        reg(Material.FIRE, new BlockDropless());
-        reg(Material.MONSTER_EGGS, new BlockDropless());
+        reg(Material.MELON_STEM, new BlockMelonStem()).setPlaceSound(Sound.DIG_GRASS);
+        reg(Material.NETHER_WARTS, new BlockDirectDrops(Material.NETHER_STALK)).setPlaceSound(Sound.DIG_GRASS);
+        reg(Material.POTATO, new BlockDirectDrops(Material.POTATO_ITEM)).setPlaceSound(Sound.DIG_GRASS);
+        reg(Material.PUMPKIN_STEM, new BlockPumpkinStem()).setPlaceSound(Sound.DIG_GRASS);
+        reg(Material.CROPS, new BlockDirectDrops(Material.SEEDS)).setPlaceSound(Sound.DIG_GRASS);
+        reg(Material.CAKE_BLOCK, new BlockDropless()).setPlaceSound(Sound.DIG_WOOL);
+        reg(Material.WEB, new BlockDirectDrops(Material.STRING)).setPlaceSound(Sound.DIG_STONE);
+        reg(Material.FIRE, new BlockDropless()).setPlaceSound(Sound.DIG_STONE); //Maybe
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/ItemTable.java:87]: Comment needs to be confirmed/addressed

**Comment by turt2live** [comment]
Closing due to inactivity (see [the PR handling guidelines](https://github.com/GlowstoneMC/Glowstone/wiki/PR-Handling)) and the inability to be merged. Please direct all conversation to #242.

Thank you for your contribution!

**Pull request closed without merging**